dotnet/src/Agents/Core/Chat/ChatExecutionSettings.cs

```diff
@@ -0,0 +1,60 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.SemanticKernel.Agents.Chat;
+
+/// <summary>
+/// Delegate definition for <see cref="ChatExecutionSettings.TerminationStrategy"/>.
+/// </summary>
+/// <param name="agent">The agent actively interacting with the nexus.</param>
+/// <param name="history">The chat history.</param>
+/// <param name="cancellationToken">The <see cref="CancellationToken"/> to monitor for cancellation requests. The default is <see cref="CancellationToken.None"/>.</param>
+/// <returns>True to terminate chat loop.</returns>
+public delegate Task<bool> TerminationCriteriaCallback(Agent agent, IReadOnlyList<ChatMessageContent> history, CancellationToken cancellationToken);
+
+/// <summary>
+/// Delegate definition for <see cref="ChatExecutionSettings.SelectionStrategy"/>.
+/// </summary>
+/// <param name="agents">The agents participating in chat.</param>
+/// <param name="history">The chat history.</param>
+/// <param name="cancellationToken">The <see cref="CancellationToken"/> to monitor for cancellation requests. The default is <see cref="CancellationToken.None"/>.</param>
+/// <returns>The agent who shall take the next turn.</returns>
+public delegate Task<Agent?> SelectionCriteriaCallback(IReadOnlyList<Agent> agents, IReadOnlyList<ChatMessageContent> history, CancellationToken cancellationToken);
+
+/// <summary>
+/// Settings that affect behavior of <see cref="AgentChat"/>.
+/// </summary>
+/// <remarks>
+/// Default behavior result in no agent selection.
+/// </remarks>
+public class ChatExecutionSettings
+{
+    /// <summary>
+    /// Restrict number of turns to one, by default.
+    /// </summary>
+    public const int DefaultMaximumIterations = 1;
+
+    /// <summary>
+    /// The maximum number of agent interactions for a given nexus invocation.
+    /// </summary>
+    public int MaximumIterations { get; set; } = DefaultMaximumIterations;
+
+    /// <summary>
+    /// Optional strategy for evaluating whether to terminate multiturn chat.
+    /// </summary>
+    /// <remarks>
+    /// See <see cref="TerminationStrategy"/>.
+    /// </remarks>
+    public TerminationCriteriaCallback? TerminationStrategy { get; set; }
+
+    /// <summary>
+    /// An optional strategy for selecting the next agent.
+    /// </summary>
+    /// <remarks>
+    /// See <see cref="SelectionStrategy"/>.
+    /// </remarks>
+    public SelectionCriteriaCallback? SelectionStrategy { get; set; }
```

Comment:
Why is this optional?

Should we have a default strategy set?

Reply:
There are fall-through defaults already, within the implementation of `AgentGroupChat`.  I considered creating default singleton strategies but wanted to limit code-sprawl.  On consideration of your comment, I agree that an explicit default goes a long way in discoverability.  I'll take a cycle to update...

Reply:
I do love the elegance of `null` as a default, but I've refactored so that `AgentGroupChat.ExecutionSettings` no longer defaults to null and has an explicit default.  In addition, `ChatExecutionSettings.TerminationStrategy` has also been refactored to not-null with an explicit default.

For `ChatExecutionSettings.SelectionStrategy`, it doesn't feel like we have sufficient scope to default to one behavior over another.  There's nothing about our stock `SequentialSelectionStrategy` that would make it a meaningful default (its just as likely to be _wrong_ as _correct_).  

I really struggled with this asymmetry, but it feel proper to require the caller to make a decision here.

In the absence of a selection-strategy, the agent specific invocation can always be utilized on `AgentGroupChat`:

`AgentGroupChat.InvokeAsync(Agent, CancellationToken)`

I've clarified comments around this.